bundles/storage/org.eclipse.smarthome.storage.json/src/main/java/org/eclipse/smarthome/storage/json/JsonStorage.java

```diff
@@ -186,8 +192,7 @@ private T deserialize(Map<String, Object> jsonValue) {
                 loadedValueType = (Class<T>) classLoader.loadClass((String) jsonValue.get(CLASS));
             }
 
-            String jsonString = mapper.toJson(jsonValue.get("value"));
-            value = mapper.fromJson(jsonString, loadedValueType);
+            value = innerMapper.fromJson((JsonElement) jsonValue.get(VALUE), loadedValueType);
```

Comment:
'jsonValue'  is defined in the method declaration but is actually the 'innerMap'. Its a little bit strange to have a Map called jsonValue (I comment here because its not part of your PR). Can we come up with a better name?